addons/account/models/ir_actions_report.py

```diff
@@ -29,6 +33,14 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
             attachment = invoice.message_main_attachment_id
             if attachment:
                 stream = io.BytesIO(attachment.raw)
+                if attachment.mimetype.startswith('image'):
+                    #This import is needed to make sure a PDF stream can be saved in Image
+                    from PIL import PdfImagePlugin
```

Comment:
is there a drawback to doing it in the import? (it's possible there is a drawback, for example for performance)

Reply:
@nle-odoo thank you for feedback, honestly I just copied the solution from 15 in this [commit](https://github.com/odoo/odoo/commit/b82a6cdea21a66588ddcafcdfb5026ec22546745)

Reply:
don't hesitate to indicate it in the commit if you are using code that was already in odoo, this let people know that the code has been running for several years and doesn't seem to cause issue.

the code you are linking was removed in a refactoring: 

https://github.com/odoo/odoo/commit/7c4e591d58dd2b882f0c237d52490b2756c94203

With the fact that we are copying code from 15.0 to 16.0, this looks good to me then.

Reply:
From the comment, it looks like it should be in a try/except though.
It seems like a check on the version of PIL, which could be done in other, cleaner ways...

Reply:
@william-andre in case of exception what is the best option ? raise an error or ignore this attachment ?

Reply:
this file is in pillow at least since pillow first release 2012

I had not checked in details, but I thought this was to load some optional module of a library. @abla001 could check  if it works without it if we keep this solution.

Reply:
@nle-odoo I tested it without importing PIL and it breaks it gave an error `KeyError: 'PDF'`

Reply:
Could you give the entire traceback?

Reply:
@william-andre 
```
Traceback (most recent call last):
  File "/home/odoo/16/odoo/addons/web/controllers/report.py", line 116, in report_download
    response = self.report_routes(reportname, docids=docids, converter=converter, context=context)
  File "/home/odoo/16/odoo/odoo/http.py", line 698, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/home/odoo/16/odoo/addons/web/controllers/report.py", line 45, in report_routes
    pdf = report.with_context(context)._render_qweb_pdf(reportname, docids, data=data)[0]
  File "/home/odoo/16/odoo/addons/account/models/ir_actions_report.py", line 64, in _render_qweb_pdf
    return super()._render_qweb_pdf(report_ref, res_ids=res_ids, data=data)
  File "/home/odoo/16/odoo/odoo/addons/base/models/ir_actions_report.py", line 793, in _render_qweb_pdf
    collected_streams = self._render_qweb_pdf_prepare_streams(report_ref, data, res_ids=res_ids)
  File "/home/odoo/16/odoo/addons/account_edi_ubl_cii/models/ir_actions_report.py", line 58, in _render_qweb_pdf_prepare_streams
    collected_streams = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids=res_ids)
  File "/home/odoo/16/odoo/addons/account_edi/models/ir_actions_report.py", line 14, in _render_qweb_pdf_prepare_streams
    collected_streams = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids=res_ids)
  File "/home/odoo/16/odoo/addons/account/models/ir_actions_report.py", line 40, in _render_qweb_pdf_prepare_streams
    img.convert("RGB").save(output_stream, format="pdf")
  File "/home/odoo/.local/lib/python3.8/site-packages/PIL/Image.py", line 2153, in save
    save_handler = SAVE[format.upper()]
KeyError: 'PDF'
```